src/github.com/getlantern/enproxy/proxy.go

```diff
@@ -169,11 +175,15 @@ func (p *Proxy) handleWrite(resp http.ResponseWriter, req *http.Request, lc *laz
 		badGateway(resp, fmt.Sprintf("Unable to write to connOut: %s", err))
 		return
 	}
-	if p.Host != "" {
-		// Always feed this so clients will be guaranteed to reach
-		// this particular proxy even if they originally reached us
-		// through (e.g.) DNS round robin.
-		resp.Header().Set(X_ENPROXY_PROXY_HOST, p.Host)
+	host := ""
+	if p.HostFn != nil {
+		host = p.HostFn(req)
+	}
+	if host == "" {
+		host = p.Host
+	}
+	if host != "" {
```

Comment:
@myleshorton: I've just added some comments to the actual PR:

https://github.com/getlantern/flashlight-build/commit/6feb9b51b62abc3a40f14121c871d1eeb72bd893